scripts/error-codes/dev-expression-with-codes.js

```diff
@@ -0,0 +1,179 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+var evalToString = require('./evalToString');
+var existingErrorMap = require('./codes.json');
+var invertObject = require('./invertObject');
+var prodInvariantName = require('./constants').prodInvariantName;
+
+var errorMap = invertObject(existingErrorMap);
+var prodInvariantModuleName = 'reactProdInvariant';
+
+module.exports = function(babel) {
+  var t = babel.types;
+
+  var SEEN_SYMBOL = Symbol();
+
+  var buildRequire = babel.template(`var IMPORT_NAME = require(SOURCE);`);
+
+  var REQUIRE_PROD_INVARIANT = buildRequire({
+    IMPORT_NAME: t.identifier(prodInvariantName),
```

Comment:
I think we should consider doing this differently. Right now we don't have a guarantee that this identifier doesn't already exist (practically it probably won't matter but just in case…). We can use babel helpers to generate a unique identifier to use though and it'll be tracked on a per-file basis. This is how I did it for the object-assign transform: https://github.com/facebook/fbjs/blob/7c77c3c2ebcf0f1174512a66a08998dcff3d4cc5/babel-preset/plugins/object-assign.js#L15-L27